spec/index.bs

```diff
@@ -4183,41 +4189,71 @@ has a default layout created and used instead.
 
 ### <dfn dictionary>GPUProgrammableStage</dfn> ### {#GPUProgrammableStage}
 
+A {{GPUProgrammableStage}} describes the entry point in the user-provided
+{{GPUShaderModule}} that controls one of the programmable stages of a [=pipeline=].
+
 <script type=idl>
 dictionary GPUProgrammableStage {
     required GPUShaderModule module;
     required USVString entryPoint;
+    record<USVString, double> constants;
 };
 </script>
 
-A {{GPUProgrammableStage}} describes the entry point in the user-provided
-{{GPUShaderModule}} that controls one of the programmable stages of a [=pipeline=].
+<dl dfn-for=GPUProgrammableStage dfn-type=dict-member>
+    : <dfn>constants</dfn>
+    ::
+        Specifies the values of [=pipeline-overridable=] constants which are [=statically accessed=]
+        by the entry point {{GPUProgrammableStage/entryPoint}} in the shader module
+        {{GPUProgrammableStage/module}}.
 
-<div algorithm>
-    <dfn abstract-op>validating GPUProgrammableStage</dfn>(stage, descriptor, layout)
-        **Arguments:**
-            - {{GPUShaderStage}} |stage|
-            - {{GPUProgrammableStage}} |descriptor|
-            - {{GPUPipelineLayout}} |layout|
+        Each key-value pair must uniquely identify a [=pipeline-overridable=] constant
+        (by a string representing the numeric ID of the constant, if specified;
+        otherwise, by the constant's identifier name).
```

Comment:
This section could really benefit from an example, because it took me way too long to realize that "a string representing the numeric ID of the constant..." was simply describing this:

```js
{
  1200: 2.4,
  depth: 1.0
}
```

I am curious why we feel the numeric IDs are necessary, though. Seems like the identifier names are frequently going to be the easier thing to use.

Reply:
It's primarily a porting thing. Native APIs identify these numerically, so e.g. SPIR-V shaders converting to WGSL would benefit. Metal also allows you to set them by name, but it's just doing that with a lookup from name to number.

What we've arrived on here *always* identifies them by a string, but you can override the string (currently only to a numeric override, but in the future you could potentially override it to a different string).

Reply:
Example added

Reply:
The example helps a lot, thanks! Though the way you phrased the comment makes me want to clarify: If you provide an ID then the _only_ way you can set the constant is via the numeric ID, right?

Reply:
That's the proposal in this PR, yeah.